front/src/common/BootstrapSNCF/FormSNCF/DebouncedNumberInputSNCF.tsx

```diff
@@ -28,19 +32,27 @@ const DebouncedNumberInputSNCF = ({
   }, [input]);
 
   const checkChangedInput = (newValue: number | null) => {
-    if (newValue !== null && newValue !== input && min <= newValue && newValue <= max)
+    if (
+      newValue !== null &&
+      newValue !== input &&
+      min <= newValue &&
+      (max === undefined || newValue <= max)
+    ) {
       setInput(newValue);
-    else if (value === null && input !== 0) setInput(0);
+    } else if (value === null && input !== 0) {
+      const previousValue = input;
+      setInput(previousValue);
+    }
   };
 
   useDebouncedFunc(value, debouncedDelay, checkChangedInput);
 
   return (
-    <div className="debounced-number-input">
+    <div className={`${showFlex ? 'debounced-number-input' : ''}`}>
```

Comment:
You can also use cx :)
```suggestion
    <div className={`${showFlex && 'debounced-number-input' }`}>
```